### PR TITLE
Driver-filterable logging plus OI prune

### PR DESCRIPTION
**Pull request by phroa**
The robot code must be restarted before changes take effect, but they work.

### CODE REVIEW

**Comment by binnur** [comment]
I did a quick look over -- logic + code looks cleaner. Nice job! :)

**Comment by dbadb** [comment]
Jack - i integrated your intake command pull request and this triggered conflicts with this change.  Thoughts?  

Also: i notice that in OI this check creeped back in:

if (m_robot.getIntake().initialized())..

In my mind, this shouldn't been needed as long as:

1. we have Initialized() checks in every public method that can be invoked by commands
2. commands don't somehow reach-around these methods..

File: src/org/usfirst/frc/team4915/steamworks/OI.java [hunk 1]
```diff
@@ -66,24 +68,23 @@ public OI(Robot robot)
         }
     }
 
+    public Command getAutoCommand()
+    {
+        return null;
```
Comment (phroa) [inline src/org/usfirst/frc/team4915/steamworks/OI.java:73]: I don't think the error I was observing earlier was actually due to this, patched to return null now.

**Comment by dbadb** [comment]
Jack: one last comment.  I see that you are reading the value in the smart dashboard effectively during the OI constructor.   Since this occurs during robotInit, the question is: are you certain that the networktable transport has occurred in all startup conditions?   Another way to ask the same question:  does robotInit occur "late enough" to get the "right" value for the debugging levels?

**Comment by phroa** [comment]
We'll need to do some more testing for "all startup conditions", but it works for me so far. SmartDashboard seems to be fully set up before OI's constructor.

**Pull request merged**